### PR TITLE
[v1.15.x] Backport Extractor Regex Replace functionality

### PR DESCRIPTION
**Pull request by ben-taussig-solo**
# Description
 - Backport commit 128d2432bc2633a605380c897f117e3544093613 to the one-fifteen branch.

## Changes unique to this backport
  - had to rename `testServerVs` to `testRunnerVs` in gateway kube2e tests https://github.com/solo-io/gloo/pull/9235/commits/8ecc38c8b0d42429216b8331248138c4c87917b7
  - updated changelog
  - Updated a line in the user guide to indicate that this functionality is available in 1.15, instead of 1.17: https://github.com/solo-io/gloo/commit/f451ccf9b087f9e6596ceaa16a5f96e0e1be4d4c
  - otherwise, the changes in this PR are identical to those in the PR against main

### CODE REVIEW

File: docs/content/guides/traffic_management/request_processing/transformations/_index.md [hunk 1]
```diff
@@ -270,6 +270,76 @@ Extracted values can be used in two ways:
   - If `advancedTemplates` is set to `true`, format the string such as `{{ extraction(my-extractor) }}`.
 - You can use them in conjunction with the `mergeExtractorsToBody` body transformation type to merge them into the body.
 
+##### Extractor Modes
+
+Starting with Gloo Edge 1.17, extractors support multiple different modes of operation. These modes enable more powerful and flexible extraction capabilities. You can specify the mode of operation for an extractor using the `mode` field. The supported modes are:
```
Comment (jbohanon) [inline docs/content/guides/traffic_management/request_processing/transformations/_index.md:275]: This is not correct here, and if it merges, won't be correct on the main branch either

Comment (ben-taussig-solo) [reply docs/content/guides/traffic_management/request_processing/transformations/_index.md:275]: Ah good catch, thanks for pointing this out

Comment (ben-taussig-solo) [reply docs/content/guides/traffic_management/request_processing/transformations/_index.md]: Updated this to say `1.15` here: https://github.com/solo-io/gloo/pull/9235/commits/f451ccf9b087f9e6596ceaa16a5f96e0e1be4d4c

Comment (jbohanon) [reply docs/content/guides/traffic_management/request_processing/transformations/_index.md]: Can we put the patch v also?

Comment (ben-taussig-solo) [reply docs/content/guides/traffic_management/request_processing/transformations/_index.md]: Also put up a PR against main here: https://github.com/solo-io/gloo/pull/9239

**Review by jbohanon — APPROVED** [review]

File: projects/gloo/api/external/envoy/extensions/transformation/transformation.proto [hunk 2]
```diff
@@ -159,16 +176,41 @@ message Extraction {
...
+  //   of the replacement will be the source itself.
+  // - In REPLACE_ALL mode, the regex is applied repeatedly to find all occurrences within the source that match.
+  //   Each matching occurrence is replaced with the value in `replacement_text`. In this mode, the configuration is rejected
+  //   if `subgroup` is set. If the regex does not match the source, the result of the replacement will be the source itself.
   string regex = 2;
 
-  // If your regex contains capturing groups, use this field to determine which
-  // group should be selected.
+  // If your regex contains capturing groups, use this field to determine the
+  // group that you want to seelc. Defaults to 0.
```
Comment (bewebi) [inline projects/gloo/api/external/envoy/extensions/transformation/transformation.proto:192]:
```suggestion
  // group that you want to select. Defaults to 0.
```

File: projects/gloo/api/v1/options/transformation/transformation.proto [hunk 3]
```diff
@@ -110,16 +127,41 @@ message Extraction {
...
+  //   of the replacement will be the source itself.
+  // - In REPLACE_ALL mode, the regex is applied repeatedly to find all occurrences within the source that match.
+  //   Each matching occurrence is replaced with the value in `replacement_text`. In this mode, the configuration is rejected
+  //   if `subgroup` is set. If the regex does not match the source, the result of the replacement will be the source itself.
   string regex = 2;
 
-  // If your regex contains capturing groups, use this field to determine which
-  // group should be selected.
+  // If your regex contains capturing groups, use this field to determine the
+  // group that you want to seelc. Defaults to 0.
```
Comment (bewebi) [inline projects/gloo/api/v1/options/transformation/transformation.proto:143]:
```suggestion
  // group that you want to select. Defaults to 0.
```

**Review by bewebi — COMMENTED** [review]
One typo in two spots, otherwise LGTM

File: projects/gloo/api/external/envoy/extensions/transformation/transformation.proto (hunk 2 above)
Comment (ben-taussig-solo) [reply projects/gloo/api/external/envoy/extensions/transformation/transformation.proto]: Resolved with: https://github.com/solo-io/gloo/pull/9235/commits/d29e61b452d9d5f9e695abef20bd90bf213e550d

File: projects/gloo/api/v1/options/transformation/transformation.proto (hunk 3 above)
Comment (ben-taussig-solo) [reply projects/gloo/api/v1/options/transformation/transformation.proto]: Resolved with: https://github.com/solo-io/gloo/pull/9235/commits/d29e61b452d9d5f9e695abef20bd90bf213e550d

**Review by bewebi — APPROVED** [review]

**Review by jbohanon — APPROVED** [review]

**Pull request merged**